scripts/sell_unused_slots.py

```diff
@@ -0,0 +1,125 @@
+from datetime import (
+    datetime,
+    timedelta,
+)
+import sys
+from typing import (
+    Dict,
+    Iterable,
+    List,
+)
+
+from azul import (
+    cached_property,
+    logging,
+)
+from azul.bigquery_reservation import (
+    SlotManager,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.modules import (
+    load_app_module,
+)
+
+log = logging.getLogger(__name__)
+
+
+class ReindexDetector:
+    # Minutes
+    interval = 5
+
+    # Minimum number of contribution Lambda invocations per interval for a
+    # reindexing to be considered active
+    threshold = 0
+
+    @cached_property
+    def _cloudwatch(self):
+        return aws.client('cloudwatch')
+
+    @cached_property
+    def _lambda(self):
+        return aws.client('lambda')
+
+    @cached_property
+    def contribution_lambda_suffix(self) -> str:
+        sep = '-'
+        indexer = load_app_module('indexer')
+        return sep + indexer.contribute.lambda_name
+
+    def is_reindex_active(self) -> bool:
+        function_names = self.list_contribution_lambdas()
+        invocations_by_function = self._lambda_invocation_counts(function_names)
+        reindex_active = False
+        for function_name, invocation_count in invocations_by_function.items():
+            description = (f'{function_name}: {invocation_count} invocations in '
+                           f'the last {self.interval} minutes')
+            if invocation_count > self.threshold:
+                log.info(f'Active reindex for {description}')
+                reindex_active = True
+            else:
+                log.debug(f'No active reindex for {description}')
+        return reindex_active
+
+    def list_contribution_lambdas(self) -> List[str]:
+        """
+        Search Lambda functions for the names of contribution Lambdas.
+        """
+        contribution_lambdas = []
+        paginator = self._lambda.get_paginator('list_functions')
+        for response in paginator.paginate():
+            for function in response['Functions']:
+                function_name = function['FunctionName']
+                if function_name.endswith(self.contribution_lambda_suffix):
```

Comment:
azul.Config has unqualified_resource_name, see if you can use that rather than hard-coding the separator or assuming a suffix match.